internal/testplanet/uplink_test.go

```diff
@@ -37,3 +40,72 @@ func TestUploadDownload(t *testing.T) {
 
 	assert.Equal(t, expectedData, data)
 }
+
+func TestDownloadWithSomeNodesOffline(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 5, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+
+		// first, upload some remote data
+		ul := planet.Uplinks[0]
+		satellite := planet.Satellites[0]
+
+		testData := make([]byte, 1*memory.MiB)
+		_, err := rand.Read(testData)
+		require.NoError(t, err)
+
+		err = ul.UploadWithConfig(ctx, satellite, &uplink.RSConfig{
+			MinThreshold:     2,
+			RepairThreshold:  3,
+			SuccessThreshold: 4,
+			MaxThreshold:     5,
+		}, "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		// get a remote segment from pointerdb
+		pdb := satellite.Metainfo.Service
+		listResponse, _, err := pdb.List("", "", "", true, 0, 0)
+		require.NoError(t, err)
+
+		var path string
+		var pointer *pb.Pointer
+		for _, v := range listResponse {
+			path = v.GetPath()
+			pointer, err = pdb.Get(path)
+			require.NoError(t, err)
+			if pointer.GetType() == pb.Pointer_REMOTE {
+				break
+			}
+		}
+
+		// calculate how many storagenodes to kill
+		redundancy := pointer.GetRemote().GetRedundancy()
+		remotePieces := pointer.GetRemote().GetRemotePieces()
+		minReq := redundancy.GetMinReq()
+		numPieces := len(remotePieces)
+		toKill := numPieces - int(minReq)
+
+		nodesToKill := make(map[storj.NodeID]bool)
+		for i, piece := range remotePieces {
```

Comment:
couldn't you just kill the first toKill nodes rather than making this map?

Reply:
No, because StopPeer needs to take a planet.Peer as an argument, which we can't access in this loop.

Reply:
maybe this?
for i:=0; i<toKill; i++{
planet.StopPeer(planet.StorageNodes[i].Identity.ID)

Reply:
are planet.StorageNodes's peers?  EG:
for i:=0; i<toKill; i++{
err = planet.StopPeer(planet.StorageNodes[i])

Reply:
the problem with this is that not all storage nodes in the test get uploaded to, and for this test, we specifically want to kill storage nodes that are storing pieces (as we found from looking in the pointer).